src/functional/ffront/foast_passes/utils.py

```diff
@@ -0,0 +1,48 @@
+from functional.ffront import field_operator_ast as foast
+
+
+def compute_assign_indices(
+    targets: list[foast.FieldSymbol | foast.TupleSymbol | foast.ScalarSymbol | foast.Starred],
+    num_elts: int
+) -> list[int | tuple]:
+    """
+    Computes a list of relative indices, mapping each target to its respective value(s).
+
+    This function is used when mapping a tuple of targets to a tuple of values, and also handles the
+    case in which tuple unpacking is done using a Starred operator.
+
+    Examples
+    --------
+    The below are examples of different types of unpacking and the
+    generated indices. Note that the indices in the tuple correspond
+    to the lower and upper slice indices of a Starred variable.
+
+    ..  code-block:: python
+
+        # example pseudocode
+        a, *b, c = (1, 2, 3, 4)
+        (0, (1, -1), -1)
+
+        *a, b, c = (1, 2, 3, 4)
+        [(0, -2), -2, -1]
+
+        a, b, *c = (1, 2, 3, 4)
+        [0, 1, (2, 5)]
+    """
+    indices = list(range(len(targets)))
+
+    for idx, elt in enumerate(targets):
+        if isinstance(elt, foast.Starred):
+            break
+        indices[idx] = idx
+
+    for idx, elt in reversed(list(enumerate(targets))):
+        rel_idx = idx - len(targets)
+        if isinstance(elt, foast.Starred):
+            star_lower, star_upper = max(indices), min(indices)
+            if star_upper == 0:
+                star_upper = num_elts
+            indices[idx] = (star_lower, star_upper)
+            break
+        indices[idx] = rel_idx
+    return indices
```

Comment:
```suggestion
    starred_lower, starred_upper = None, None

    for idx, elt in enumerate(targets):
        if isinstance(elt, foast.Starred):
            starred_lower = idx
            break

    for idx, elt in zip(reversed(range(num_elts)), reversed(targets)):
        if isinstance(elt, foast.Starred):
            starred_upper = idx + 1
            break

    if starred_lower:
        return list(range(0, starred_lower)) \
               + [(starred_lower, starred_upper)] \
               +  list(range(starred_upper, num_elts))
    return list(range(0, num_elts))  # no starred target
```
To be honest I have a hard time digesting how this function works, in particular the `star_lower, star_upper = max(indices), min(indices)` in conjunction with what is inside `indices` at that point. I would suggest this version of the algorithm that clearly separates between the computation of the starred indices and the construction of the final result.

Reply:
Very nice! This is definitely cleaner. I had to change a small check at the end from:

`if starred_lower:` to `if starred_lower is not None:` other than that it works!